[Reviewer: GDR] Don't let Chronos run as the root user.

sathiyan-sivathas opened the pull request:
Copied other examples to see how this should be done.

add --chuid when starting Chronos.
add --user when stopping it.

graemerobertson commented:
For bug fixes we normally reference the issue in the PR description for 2 reasons - firstly to give context for the fix, and secondly because github will parse things like "fixes #197" and close the issue automatically once the PR is merged.

graemerobertson commented:
This looks good but I think there's a couple of bits missing.
* I think we need to create the Chronos user, like we do in, for example, [Homestead](https://github.com/Metaswitch/homestead/blob/178fdcbf0f4898b8890fd03ff2b9a09005c2c588/debian/homestead.postinst#L70).
* I think we also need to be careful that on upgrade (from a Chronos running as the root user to a Chronos running as the chronos user), Chronos can still write to all the relevant files. For example, I suspect that it will no longer be able to write to the log directory. We currently create the empty log directory using a debian mechanism (see debian/dirs) and it will be owned by root I think, but perhaps we should now be again replicating the logic in, for example, [Homestead](https://github.com/Metaswitch/homestead/blob/178fdcbf0f4898b8890fd03ff2b9a09005c2c588/debian/homestead.postinst#L74).

We should also test this change - I think it's probably necessary and sufficient to just test upgrade so let's discuss how best to do that tomorrow.

sathiyan-sivathas commented:
I've made the changes you suggested.  Could you check them and then can we discuss the testing.

graemerobertson commented:
This looks good, and I think we can also now delete https://github.com/Metaswitch/chronos/blob/dev/debian/dirs since this is no longer necessary with your change (the dirs file in the debian directory specifies a list of empty directories to create on install - /var/log/chronos is now created in the postinst instead).

eleanor-merry commented on debian/chronos.preinst:
```diff
@@ -0,0 +1,74 @@
+# @file chronos.preinst
+#
+# Project Clearwater - IMS in the Cloud
+# Copyright (C) 2015  Metaswitch Networks Ltd
```
Copyright year...

sathiyan-sivathas commented:
I've moved the Chronos PID to it's own directory and added a preinst script.  I've left the code for stopping Chronos in the prerm script just in case we decide we can remove the preinst script at a later date.  Please re-review.

graemerobertson commented on debian/chronos.preinst:
```diff
@@ -0,0 +1,74 @@
+NAME=chronos
+
+case "$1" in
+    install|upgrade)
```
We only want to do this on upgrade - you should move install into the no-op below with abort-upgrade.

graemerobertson commented on debian/chronos.preinst:
```diff
@@ -0,0 +1,74 @@
+
+case "$1" in
+    install|upgrade)
+        # We might have moved Chronos PID file from /var/run/chronos.pid to
```
I think you should be a bit more explicit - i.e. "At some point we moved the Chronos PID file from blah to blah, and if this upgrade is moving from pre-move to post-move then we need to do this".

(Write something more eloquent than that though obviously)

graemerobertson commented:
Looks good, a couple of minor comments
